### PR TITLE
Upgrade of `commons-fileupload2-jakarta` to support Jakarta Servlet API 6.x

### PR DESCRIPTION
**Pull request by ideas-into-software**
Done as a prerequisite of upgrading Apache Felix to Jakarta Servlet API 6.x. 

Please see https://issues.apache.org/jira/browse/FELIX-6612 and https://github.com/apache/felix-dev/pull/218 for more info.

### CODE REVIEW

**Comment by garydgregory** [comment]
@ideas-into-software 
Thank you for your PR.
Git master now contains a Servlet 6-specific module.
Please test your use case. 
Git master does _not_ contain the kind of OSGi changes in this PR. If you care about those, please rebase on master.
TY!

**Comment by ideas-into-software** [comment]
@garydgregory 

Hi. As you know, Jakarta Servlet API 6.x support was not in `master` when I submitted this PR, hence my PR. 

I see you refactored and created two separate modules, i.e. for JSA5 and JSA6.

It would useful to keep the enhanced OSGi metadata, so I'll re-submit updated PR.

Thank you!

> @ideas-into-software Thank you for your PR. Git master now contains a Servlet 6-specific module. Please test your use case. Git master does _not_ contain the kind of OSGi changes in this PR. If you care about those, please rebase on master. TY!

File: commons-fileupload2-jakarta/pom.xml [hunk 1]
```diff
@@ -26,6 +26,7 @@
   </parent>
 
   <artifactId>commons-fileupload2-jakarta</artifactId>
+  <version>3.0.0-SNAPSHOT</version>
```
Comment (garydgregory) [inline commons-fileupload2-jakarta/pom.xml:29]: The version is inherited.

**Comment by garydgregory** [comment]
👍

**Comment by ideas-into-software** [comment]
Closing as new PR was just submitted: https://github.com/apache/commons-fileupload/pull/233

**Pull request closed without merging**